api/src/main/java/org/commonjava/indy/data/DefaultStoreValidator.java

```diff
@@ -0,0 +1,195 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.*;
+
+
+public class DefaultStoreValidator implements StoreValidator {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DefaultStoreValidator.class);
+
+    @Inject
+    @WeftManaged
+    @ExecutorConfig( named="store-validation", threads=2, priority=6 )
+    private ExecutorService executorService;
+
+    @Inject
+    IndyConfiguration configuration;
+
+    @Override
+    public ArtifactStoreValidateData validate(ArtifactStore artifactStore) throws InvalidArtifactStoreException, MalformedURLException {
+        final CountDownLatch httpRequestsLatch = new CountDownLatch(2);
+        final HashMap<String, String> errors = new HashMap<>();
+        Optional<URL> remoteUrl = Optional.empty();
+
+
+        try {
+            if(StoreType.remote == artifactStore.getType()) {
+
+                // Cast to Remote Repository
+                RemoteRepository remoteRepository = (RemoteRepository) artifactStore;
+
+                //Validate URL from remote Repository URL
+                remoteUrl = Optional.of(new URL(remoteRepository.getUrl()));
+                // Check if remote.ssl.required is set to true and that remote repository protocol is https = throw IndyArtifactStoreException
+                if(configuration.isSSLRequired() && !remoteUrl.get().getProtocol().equalsIgnoreCase("https")) {
+
+                    //Check First if this remote repository is in allowed repositories from remote.nossl.hosts Config Variable
+                    List<String> remoteNoSSLHosts = configuration.getRemoteNoSSLHosts();
+                    String host = remoteUrl.get().getHost();
+                    for(String remoteHost : remoteNoSSLHosts) {
+                        if(!remoteHost.contains(host)) {
+                            errors.put(NON_SSL, "Not Valid SSL Remote Repository");
+                            LOGGER.warn("NON-SSL RemoteRepository with URL: " + remoteUrl);
+                            // throw new InvalidArtifactStoreException("Not Valid SSL Remote Repository", null, null);
+                        }
+                    }
+
+                }
+                // Execute HTTP GET & HEAD requests in separate thread pool from executor service
+                Future<Integer> httpGetStatus = executeGetHttp(new HttpGetTask(new HttpGet(remoteUrl.get().toURI())),httpRequestsLatch);
+                Future<Integer> httpHeadStatus = executeHeadHttp(new HttpHeadTask(new HttpHead(remoteUrl.get().toURI())),httpRequestsLatch);
+                // Waiting for Http GET & HEAD Request Executor tasks to finish
+                httpRequestsLatch.await();
+                errors.put(HTTP_GET_STATUS, httpGetStatus.get().toString());
```

Comment:
Yes , it will insert returned Http Status code for GET request , i was thinking that it would be good thing if we have those values for our http requests on remote repositories , because sometimes maybe return status code will be redirected ( 302 ) or something like that.